python/ml4ir/applications/ranking/model/metrics/metrics_impl.py

```diff
@@ -219,3 +219,27 @@ def __init__(
         **kwargs
     ):
         super(CategoricalAccuracy, self).__init__(name=name)
+
+
+class TopKCategoricalAccuracy(metrics.TopKCategoricalAccuracy):
+    """
+    Custom metric class to compute the Top K Categorical Accuracy.
+
+    Currently a wrapper around tf.keras.metrics.TopKCategoricalAccuracy that
+    squeezes one dimension.
+    It maintains consistency of arguments to __init__
+    """
+
+    def __init__(
+        self,
+        feature_config: FeatureConfig,
+        metadata_features: Dict,
+        name="top_k_categorical_accuracy",
+        state=MetricState.NEW,
+        **kwargs
+    ):
+        super(TopKCategoricalAccuracy, self).__init__(name=name)
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        return super(TopKCategoricalAccuracy, self).update_state(
```

Comment:
@balikasg Add a comment here for why you need to do this. Basically input data shape and expected data shape by metric.

Reply:
yes, I am doing this now.